WORKSPACE

```diff
@@ -52,82 +47,66 @@ maven_jar(
     sha1 = "b6ad12d98295a7d17b3fe4b8969d0f7905626b30",
 )
 
-git_repository(
+http_archive(
     name = "io_bazel_rules_sass",
-    remote = "https://github.com/bazelbuild/rules_sass.git",
-    tag = "0.0.1",
+    sha256 = "d39d40c39a0fa2c7d05230ccf95aac3628936e4e76c0379ad324ff0b8488160f",
+    strip_prefix = "rules_sass-0.0.1",
+    url = "https://github.com/bazelbuild/rules_sass/archive/0.0.1.tar.gz",
 )
 
 load("@io_bazel_rules_sass//sass:sass.bzl", "sass_repositories")
 
 sass_repositories()
 
-git_repository(
+http_archive(
     name = "io_bazel_skydoc",
-    commit = "c57ff682364dbb1ae808b769f9e3add77cdbfad1",
-    remote = "https://github.com/bazelbuild/skydoc.git",
+    sha256 = "256bf8b64269d21fd46b8696007b5b9ef10070d79c106e74fb37979c04b6d519",
+    strip_prefix = "skydoc-c57ff682364dbb1ae808b769f9e3add77cdbfad1",
+    url = "https://github.com/bazelbuild/skydoc/archive/c57ff682364dbb1ae808b769f9e3add77cdbfad1.tar.gz",
 )
 
 load("@io_bazel_skydoc//skylark:skylark.bzl", "skydoc_repositories")
 
 skydoc_repositories()
 
-http_file(
-    name = "org_chromium_chromedriver_linux",
-    sha256 = "0c01b05276da98f203dc7eb4236c2ee7fe799b432734e088549bd0aadc71958e",
-    url = "http://chromedriver.storage.googleapis.com/2.24/chromedriver_linux64.zip",
-)
-
-http_file(
-    name = "com_google_chrome_linux",
-    sha256 = "6e26d74fd814c38cd419d1ffe87b3e81ad6cfe453e27c7a672ab3c452968e71d",
-    url = "https://commondatastorage.googleapis.com/chrome-unsigned/desktop-5c0tCh/53.0.2785.116/precise64/chrome-precise64.zip",
-)
-
-http_file(
-    name = "org_chromium_chromedriver_mac",
-    sha256 = "d4f6e13d88ecf20735138f16ab1545e855a42bce41bebe73667a028871777790",
-    url = "http://chromedriver.storage.googleapis.com/2.24/chromedriver_mac64.zip",
-)
-
-http_file(
-    name = "com_google_chrome_mac",
-    sha256 = "84b3cf4f7a9f85fa90dda837b1e38820c83c383fcb6346bbec6448d5128dd7f9",
-    url = "https://commondatastorage.googleapis.com/chrome-unsigned/desktop-5c0tCh/53.0.2785.116/mac64/chrome-mac.zip",
-)
-
-http_file(
-    name = "org_mozilla_firefox_linux",
-    sha256 = "95884070af8870a550ef70600793b6e6d5207f34af24f8b437b6c67b095e5517",
-    url = "https://ftp.mozilla.org/pub/firefox/releases/49.0/firefox-49.0.linux-x86_64.sdk.tar.bz2",
-)
+load("//web/internal:platform_http_file.bzl", "platform_http_file")
 
-http_file(
-    name = "org_mozilla_firefox_mac",
-    sha256 = "c068696c69af2da2b916e33e93755f7dda478fa6e9d17a60643cf2009bbaf8e2",
-    url = "https://ftp.mozilla.org/pub/firefox/releases/49.0/firefox-49.0.mac-x86_64.sdk.tar.bz2",
+platform_http_file(
```

Comment:
I actually want users of rules_web to specify their own download locations/versions for browsers because:
1. (minor) It makes it easier to limit what is downloaded to just the browsers that are being used (requirement 6 in your list of requirements).
2. (major) It separates the process of upgrading to new versions of rules_web (which eventually should hopefully be rare) from the process of upgrading to new versions of browsers (which for Chrome and Firefox occurs ~ every 6 weeks).